ERCS/erc-XXXX.md

```diff
@@ -13,60 +13,64 @@ requires: 712, 1193
 
 ## Abstract
 
-A new `secureRequest` method in addition to EIP-1193 `request` for dApp to interact with extension wallets in a tamperproof way, preventing any MITM from modifying the payload sent to the wallet.
-This will allow for dApps to sign their interaction with EIP-1193, and for wallet to visibly display secure interaction, therefor improving the end user experience by explicitly displaying safety
-visual cues such as a padlock.
+Introduces a new RPC method `wallet_signedRequest` that enables dapps to interact with wallets in a tamperproof manner via "signed requests".
+To send a signed request, the dapp first publishes a set of public keys via its DNS records.
+Then, the dapp signs its plaintext RPC request with a corresponding private key,
+and submits the original request payload, signature, and public key id as the parameters of `wallet_signedRequest`.
+Upon receiving a signed request, the wallet retrieves the public key from the dapp's DNS records, verifies the signature, and finally processes the plaintext request as normal.
+The wallet can subsequently indicate to the user that the request was not tampered with.
 
 ## Motivation
 
-The primary motivation for this standard is to enhance the end user experience by ensuring that they can feel safe when interacting with a trusted dApp (this is in essence very similar to HTTPS vs HTTP).
+This standard aims to enhance the end user's experience by granting them confidence that requests from their dapps have not been tampered with.
+In essence, this is similar to how HTTPS is used in the web.
 
-Currently, the communication channel between a dApp and a wallet is vulnerable to man in the middle attacks: an attacker can intercept the communication by injecting javascript code in the page, and subsequently
-modify the call data sent by the dApp to the wallet. XSS vulnerability or malicious extension both provide ways for an attacker to inject javascript. Despite EIP-720 increasing the transparency of the data being
-signed by the wallet, the average user will in effect rarely confirm the actual content of a transaction because:
+Currently, the communication channel between dapps and wallets is vulnerable to man in the middle attacks.
+Specifically, attackers can intercept RPC requests by injecting JavaScript code in the page,
+via e.g. an XSS vulnerability or due to a malicious extension.
+Once an RPC request is intercepted, it can be modified in a number of pernicious ways:
 
-* they have trust in specific well known dApp and will not read the detail thoroughly
-* the content of the transaction is compressed to save gas and therefore unreadable
-* the user is victim of social engineering
+- Editing the calldata in order to siphon funds or otherwise change the transaction outcome
+- Modify the parameters of an EIP-712 request
+- Obtain a replayable signature from the wallet
```

Comment:
Just to mitigate my own ignorance: Do you have any concrete examples of this attack?

Reply:
as in, have real life example ? No, but it would certainly be possible to get a signature without submitting it onchain. The delay could be useful in a to cover tracks. Eg, i get your signature on uniswap, then I wait a little (not too long or the nonce may change), then I execute the malicious TX at a time where you were not using any dApp, making tracking the source a little more difficult